packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/hooks.ts

```diff
@@ -0,0 +1,78 @@
+import { RefObject, useEffect, useState } from 'react';
+
+import { TimeOption } from '@grafana/data';
+
+const modulo = (a: number, n: number) => ((a % n) + n) % n;
+
+/** @internal */
+export interface UseListFocusProps {
+  localRef: RefObject<HTMLUListElement>;
+  options: TimeOption[];
+}
+
+/** @internal */
+export type UseListFocusReturn = [(event: React.KeyboardEvent) => void];
+
+/** @internal */
+export const useListFocus = ({ localRef, options }: UseListFocusProps): UseListFocusReturn => {
+  const [focusedItem, setFocusedItem] = useState(0);
+
+  useEffect(() => {
+    // Reset focused item when options have changed
+    setFocusedItem(0);
+    const items = localRef.current?.querySelectorAll<HTMLInputElement>('[data-role="item"]');
+    items?.forEach((item, i) => {
+      item.tabIndex = i === 0 ? 0 : -1;
+    });
+  }, [localRef, options]);
+
+  const handleKeys = (event: React.KeyboardEvent) => {
+    const items = localRef?.current?.querySelectorAll<HTMLInputElement>('[data-role="item"]');
+    const itemsCount = items?.length ?? 0;
+
+    let newFocusedIndex = null;
+
+    switch (event.key) {
+      case 'ArrowUp':
+        event.preventDefault();
+        event.stopPropagation();
+        newFocusedIndex = modulo(focusedItem - 1, itemsCount);
+        break;
+      case 'ArrowDown':
+        event.preventDefault();
+        event.stopPropagation();
+        newFocusedIndex = modulo(focusedItem + 1, itemsCount);
+        break;
+      case 'Home':
+        event.preventDefault();
+        event.stopPropagation();
+        newFocusedIndex = 0;
+        break;
+      case 'End':
+        event.preventDefault();
+        event.stopPropagation();
```

Comment:
Maybe you could add some generic logic to this? Store the keys in an enum and check if the pressed one is in it?

Reply:
What do you mean here? 🤔

Reply:
Just so you don't have to repeat 
```
event.preventDefault();
event.stopPropagation();
``` 
in the case for `ArrowUp`, `ArrowDown`, `Home`, `End` and `Enter`